compiler/resolution/functionResolution.cpp

```diff
@@ -5908,11 +5910,17 @@ static void resolveNewSetupManaged(CallExpr* newExpr, Type*& manager) {
       // where t is e.g Owned(MyClass)
       // or for t is unmanaged(MyClass)
       if (manager == NULL) {
-        if (isManagedPtrType(type))
+        if (isManagedPtrType(type)) {
           manager = type;
-
-        if (isUnmanagedClassType(type))
+        } else if (isUnmanagedClassType(type)) {
           manager = dtUnmanaged;
+        } else if (isClass(type) && isUndecoratedClassNew(newExpr)) {
+          if (fLegacyNew == false && fDefaultUnmanaged == false) {
+            gdbShouldBreakHere();
+            USR_WARN(newExpr, "this new call changes meaning");
```

Comment:
This message doesn't seem helpful to me. Maybe describe what's missing or what the change is?